### PR TITLE
Fix audio track null duration and add defaultRef

### LINKED ISSUE #1076: Audio track current time sometimes being set to non-real number
## Description

https://sentry.io/share/issue/96afadf57a7e40acbd0fdb52a7d7d9c2/

I think this is being caused by a discrete math order of operations bug on this line: https://github.com/WordPress/openverse-frontend/blob/29bf0a0075548014a3d1280f0b56ae1f7e5f0b77/src/components/VAudioTrack/VAudioTrack.vue#L299

```
> const audioDuration = { value: null }
undefined
> const props = { audio: null }
undefined
> audioDuration.value ?? props.audio?.duration / 1e3 ?? 0
NaN
```

As you can see, if `props.audio` is null, it is still dividing `null / 1000` which is NaN. NaN isn't affected by `??` either so it doesn't default to 0.

It'd probably be best to just split this expression into multiple lines rather than keeping it a one-liner.

Also needs to be fixed in `VGlobalAudioTrack` as well.

## Reproduction
I'm not sure how to actually reproduce this

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #1076 by @sarayourfriend 

## Description

`defaultRef` is just a computed that uses some default computed value until another value is explicitly set.

This allows us to get rid of the confusing `duration` and `audioDuration` ref duplication that was happening in the audio track components by hiding what was the effective "default" in the original computed.

It also fixes the `null / 0` error.

## Testing Instructions

Check the unit tests to make sure I didn't miss any edge case.

Test the audio track, global, search results, related audio, and single result tracks, specifically from Jamendo to make sure there isn't a regression on #692.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VAudioTrack/VGlobalAudioTrack.vue [hunk 1]
```diff
@@ -107,7 +108,11 @@ export default defineComponent({
 
     const status = ref('paused')
     const currentTime = ref(0)
-    const audioDuration = ref(null)
+    const duration = defaultRef(() => {
+      if (typeof props.audio?.duration === 'number')
```
Comment (obulat) [inline src/components/VAudioTrack/VGlobalAudioTrack.vue:112]: Can this ever be `NaN`? Would it be safer to add a check for `NaN`, too?

File: test/unit/specs/composables/default-ref.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { defaultRef } from '~/composables/default-ref'
```
Comment (obulat) [inline test/unit/specs/composables/default-ref.spec.ts:1]: Nice addition, and great to have tests for it!

**Review by obulat — APPROVED** [review]
Great fix, and useful addition of the `defaultRef`

File: src/components/VAudioTrack/VGlobalAudioTrack.vue (hunk 1 above)
Comment (sarayourfriend) [reply src/components/VAudioTrack/VGlobalAudioTrack.vue:112]: `props.audio` is the Audio result object which doesn't get modified by the frontend once it's decoded.

NaN isn't a valid JSON value so I don't think it could ever end up there:

```
> JSON.stringify({'a': NaN})
'{"a":null}'
```

File: src/composables/default-ref.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * Produces a `computed` that returns a default value until
+ * an explicit value is set.
+ *
+ * @param getDefault - A function returning the default value
+ */
+export const defaultRef = <T>(getDefault: () => T) => {
+  const explicitlySet = ref<T | typeof NotSet>(NotSet)
+  return computed<T>({
+    get() {
+      if (explicitlySet.value !== NotSet) return explicitlySet.value as T
```
Comment (dhruvkb) [inline src/composables/default-ref.ts:15]: When reading this condition "not equal to not set" feels like a double negative. Can we invert the `if`?

File: src/composables/default-ref.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { computed, ref, UnwrapRef } from '@nuxtjs/composition-api'
+
+const NotSet = Symbol('NotSet')
+
+/**
+ * Produces a `computed` that returns a default value until
+ * an explicit value is set.
+ *
+ * @param getDefault - A function returning the default value
```
Comment (dhruvkb) [inline src/composables/default-ref.ts:9]: We should also document the `@returns`.

File: .eslintrc.js [hunk 5]
```diff
@@ -143,6 +143,9 @@ module.exports = {
         ],
       },
     ],
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unused-vars.md
+    'no-unused-vars': 'off',
+    '@typescript-eslint/no-unused-vars': ['error'],
```
Comment (dhruvkb) [inline .eslintrc.js:148]: I believe this can also be fixed by switching the order of
```js
extends: [
  'plugin:@typescript-eslint/recommended',
  'eslint:recommended',
  // ...
]
```
to
```js
extends: [
  'eslint:recommended',
  'plugin:@typescript-eslint/recommended',
  // ...
]
```
in the ESLint conf.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: src/composables/default-ref.ts (hunk 3 above)
Comment (sarayourfriend) [reply src/composables/default-ref.ts]: Done in https://github.com/WordPress/openverse-frontend/pull/1129/commits/bd5d8bb8d7568cf93a94ea62316401bef8b180aa

File: .eslintrc.js (hunk 5 above)
Comment (sarayourfriend) [reply .eslintrc.js]: Good call. Done in https://github.com/WordPress/openverse-frontend/pull/1129/commits/38ee96867d8dee5856a1b43702402e30742aafb2

File: src/composables/default-ref.ts (hunk 4 above)
Comment (dhruvkb) [reply src/composables/default-ref.ts:9]: That's true. I guess it's not really important when the description is so detailed, but I always write it (even if twice).

**Review by dhruvkb — APPROVED** [review]
Concerns addressed, LGTM
